packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -81,6 +93,35 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
     }
   }
 
+  // This must be guaranteed not to return a Future that fails.
+  @override
+  bool launchDevToolsInBrowser({@required List<FlutterDevice> flutterDevices}) {
+    if (!_residentRunner.supportsServiceProtocol || _devToolsLauncher == null) {
+      return false;
+    }
+    if (_devToolsLauncher.devToolsUrl == null) {
+      _logger.startProgress('Waiting for Flutter DevTools to be served...');
+      unawaited(_devToolsLauncher.ready.then((_) {
+        _launchDevToolsForDevices(flutterDevices);
+      }));
+    } else {
+      _launchDevToolsForDevices(flutterDevices);
+    }
+    return true;
+  }
+
+  void _launchDevToolsForDevices(List<FlutterDevice> flutterDevices) {
+    assert(activeDevToolsServer != null);
```

Comment:
If we have more than one device, is this going to start N browser tabs or N browser windows?

Reply:
N tabs

Reply:
makes sense